### ISSUE TITLE
Finish `flowctl-rs` -> `flowctl` transition

### ISSUE THREAD

**Issue by psFried** [estuary/flow#375]
The TLDR is that I'm proposing renaming `flowctl` to `flowctl-go`, and `flowctl-rs` to `flowctl`.

Related PRs, for context:

- #360 Introduced `flowctl-rs`, which delegates all existing `flowctl` subcommands to the Go `flowctl`
- #361 Migrated the `parser`, `network-proxy`, and `elasticsearch-schemabuilder` into the flow repo
- estuary/connectors#130 updates the connectors repo to depend on the release binaries published from the flow repo.
- #368 includes some refactoring of `flow_cli_common`

Currently, a complete flow build includes the following binaries:

```
ls .build/package/bin
etcd flowctl flowctl-rs flow-network-proxy flow-parser flow-schemalate gazette sops
```

The architecture has evolved toward a collection of binaries that each provide a different subset of functionality. This has many advantages, but the notable disadvantage is that it can confusing for users who just want to get things done. The proposed resolution is that for 99% of users, the new `flowctl` (formerly `flowctl-rs`) is all they ever need. Any commands that normal users may want to run will be provided by `flowctl`, which will delegate to other binaries as needed.

The `git` project has kind of a similar problem, in that there's _many_ different `git-*` binaries, but end users don't typically use them directly. They use the terminology of ["porcelain" and "plumbing"](https://git-scm.com/book/en/v2/Git-Internals-Plumbing-and-Porcelain) to differentiate CLIs that are primarily intended for end users, versus the lower level commands that users almost never need to interact with directly. For flow, I think a sensible position is to say that `flowctl` (the new one) represents the entirety of the "porcelain", and everything else can be considered "plumbing".

Of course if we rename `flowctl-rs` to `flowctl`, then we're faced with the question of what to rename the existing `flowctl` to. Originally, I had thought that renaming it to just `flow` would be sensible, since it sort of conforms to the pattern of the `flow-` being more for plumbing, and `flowctl` being for porcelain.

One consideration is the implications of the name on the user experience in a shell. If a user starts typing `flo` and hits the TAB key, Bash will complete it to the shortest match from all programs on the `PATH`. If a program called `flow` is in the users `PATH`, then that's what it would complete to, without the user even seeing other options. This makes running the wrong command (`flow`) seem like the easy and obvious thing, which is sure to cause confusion among new users. The alternative of naming it `flowctl-go` seems preferable because typing `flow` and hitting TAB will then show something like the following, where the correct (`flowctl`) program seems like the obvious choice:

```
flowctl flowctl-go flow-parser flow-schemalate
$ flow█
```

So I think that `flowctl-go` is the way to go for now because it balances expediency and user experience. With some more work, I think we can do better in the future, though. One way to do that is to again take a tip from the `git` project (plenty of others do this too) and simply keep plumbing commands off of the users PATH. The plumbing commands could be put in a known location and invoked with their absolute path. That seems ideal because then the names would hardly matter at all to end users, but it requires some sort of installation process to get all the binaries in the right places. This is probably worth while in the long run, since we'll certainly want to improve the installation process anyway. But I don't think it's an important near-term priority, so `flowctl-go` seems like the way to `-go` for now.

**Comment by saterus** [comment]
This seems reasonable. 

I don't think I realized we'd end up prefixing the parser and schemalate with `flow-`, but that does make some sense. My only thought there would be that a user typing `flo<TAB>` is going to get all of them and need to type `c<TAB>` to actually get to `flowctl`. If we *didn't* _prefix_ the auxiliary commands with `flow`, we'd have `flo<TAB>` correct get a user to `flowctl` directly. However, it feels weird to put them under a `flowctl-` prefix (which would be the analogous option to `git` commands). What if we named the cli tool `flow`, rather than `flowctl`?

> seems like the way to `-go` for now.

:1st_place_medal:
